crucible-llvm/src/Lang/Crucible/LLVM/MemModel.hs

```diff
@@ -1583,6 +1584,35 @@ buildDisjointRegionsAssertion sym w dest dlen src slen = do
 
   orPred sym diffBlk =<< orPred sym destfirst srcfirst
 
+-- | Build the condition that two memory regions are disjoint, using
+-- subtraction and comparison to zero instead of direct comparison (that is,
+-- 0 <= y - x instead of x <= y). This enables semiring and abstract domain
+-- simplifications. The result if false if any offset is not positive when
+-- interpreted as signed bitvector.
+buildDisjointRegionsAssertionWithSub ::
```

Comment:
Is there a reason to preserve both versions of this? Is the condition being check semantically the same?

Reply:
This version is more restrictive because it requires all the involved numbers to be positive.